lib/ex_plasma.ex

```diff
@@ -78,57 +81,64 @@ defmodule ExPlasma do
     ...>    tx_data: <<0>>,
     ...>    tx_type: 1
     ...>  }
-    iex> ExPlasma.encode(txn)
+    iex> ExPlasma.encode(txn, signed: false)
     <<248, 104, 1, 225, 160, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 238, 237, 1, 235, 148, 29, 246, 47,
       41, 27, 46, 150, 159, 176, 132, 157, 153, 217, 206, 65, 226, 241, 55, 0, 110,
       148, 46, 38, 45, 41, 28, 46, 150, 159, 176, 132, 157, 153, 217, 206, 65, 226,
       241, 55, 0, 110, 1, 128, 148, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0>>
   """
-  @spec encode(Transaction.t()) :: binary()
-  defdelegate encode(txn), to: Transaction
+  defdelegate encode(transaction, opts \\ []), to: Transaction
 
   @doc """
-  Decode the given RLP list into a Transaction.
+  Attempt to decode the given RLP list into a Transaction.
+
+  If `signed: false` is given in the list of opts, expects the underlying RLP to not contain signatures.
```

Comment:
is this specific flag really needed? is it not possible if signatures is an empty list or full?

Reply:
It is indeed the best option that I found to deal with this, I spent some time thinking about this already. We really need to be able to specify if we want to include sigs or not.

Reply:
can you elaborate _if we want to include sigs or not_? I'm not super familiar with this part of the codebase.
What does it mean "to include it". When would you include it and when wouldn't you?

Reply:
So, for example in the childchain, we receive transactions submitted externally (from the API), these transactions have been binary (and hexed) encoded with their signatures (maybe the guy even used ex_plasma to do that? :p). So we will need to decode and validate that the signatures are there and also recover the addresses from the signatures.
However there are also some cases where we want to encode transactions without their signature, like when building merkle root hash for example. We really need both.

Reply:
Thanks!